client-guide/jmap-client-guide.mdown

```diff
@@ -373,187 +440,189 @@ The `maxChanges` argument prevents methods from returning huge volumes of data i
-                ],
-                "subject": "Camping trip",
-                "date": "2014-07-24T15:04:51Z",
-                "preview": "Hey Joe. Fancy a trip out west next week? I hea…"
-            }, {
-                "id": "fm1u314",
-                "threadId": "4f512aafed75e7fb",
-                "mailboxId": "mailbox1",
-                "isUnread": false,
-                "isFlagged": false,
-                "isAnswered": false,
-                "isDraft": false,
-                "hasAttachment": false,
-                "from": [
-                    { name: "Joe Bloggs", email: "joebloggs@fastmail.fm" }
-                ],
-                "to": [
-                    { name: "Jane Doe", email: "janedoe@fastmail.fm" }
-                ],
-                "subject": "Camping trip",
-                "date": "2014-07-24T15:04:51Z",
-                "preview": "Hey Joe. Fancy a trip out west next week? I hea…"
-            }],
-            notFound: null
-        }, "call3" ],
-        [ "threadUpdates", {
-            accountId: 'user@example.com',
-            oldState: "mc1264092",
-            newState: "mc1264097",
-            changed: [ "afed75fb4f512ae7", "b4aae3925af0a0a2" ],
-            removed: []
-        }, "call4" ]
-        [ "threads", {
-            accountId: 'user@example.com',
-            state: "mc1264097",
-            list: [{
-                id: "afed75fb4f512ae7",
-                messageIds: [ "fm1u241", "fm1u316" ]
-            }, {
-                id: "b4aae3925af0a0a2",
-                messageIds: [ "fm1u315" ]
-            }],
-            notFound: null
-        }, "call4" ]
-    ]
+```json
+[
+    [ "Mailbox/changes", {
+        "accountId": "u123456",
+        "oldState": "123",
+        "newState": "124",
+        "created": [],
+        "destroyed": [],
+        "updated": [ "mailbox1" ],
+        "updatedProperties": [ "totalEmails", "unreadEmails", "totalThreads", "unreadThreads" ]
+    }, "0" ],
+    [ "Mailbox/get", {
+        "accountId": "u123456",
+        "list": [],
+        "notFound": [],
+        "state": "124"
+    }, "1" ],
+    [ "Mailbox/get", {
+        "accountId": "u123456",
+        "list": [{
+            "id": "mailbox1",
+            "totalEmails": 1426,
+            "totalThreads": 1214,
+            "unreadEmails": 4,
+            "unreadThreads": 3
+        }],
+        "notFound": [],
+        "state": "124"
+    }, "2" ],
+    [ "Email/queryChanges", {
+        "accountId": "u123456",
+        "filter": {
+            "inMailbox": "mailbox1"
+        },
+        "sort": [
+            { "property": "receivedAt", "isAscending": false }
+        ],
+        "collapseThreads": true,
+        "newQueryState": "124:0",
+        "oldQueryState": "123:0",
+        "added": [{
+            "id": "fm1u316",
+            "index": 0
+        }, {
+            "id": "fm1u315",
+            "index": 1
+        }],
+        "removed": [],
+        "upToId": "fm1u313"
```

Comment:
Remove upToId, filter, sort, collapseThreads from the response: they're not returned any more because the client already knows what they are.